smart-contracts/contracts/cl-vault/src/vault/any_deposit.rs

```diff
@@ -0,0 +1,337 @@
+            deposit_amount_in_ratio,
+            max_slippage,
+            &recipient,
+        )?;
+
+        // rest minting logic remains same
+        Ok(Response::new()
+            .add_submessage(SubMsg::reply_on_success(
+                swap_msg,
+                Replies::AnyDepositSwap.into(),
+            ))
+            .add_attribute("method", "execute")
+            .add_attribute("action", "any_deposit")
+            .add_attribute("token_in", format!("{}{}", swap_amount, token_in_denom))
+            .add_attribute("token_out_min", format!("{}", token_out_min_amount)))
+    } else {
+        // TODO: I dont like this early return here and the fact that this function is or finishing here, or going to invoke an async reply.
+        let (mint_msg, user_shares) =
+            mint_msg_user_shares(deps, &env, &deposit_amount_in_ratio, &recipient)?;
+
+        Ok(Response::new()
+            .add_attribute("method", "execute")
+            .add_attribute("action", "any_deposit")
+            .add_attribute("amount0", deposit_amount_in_ratio.0)
+            .add_attribute("amount1", deposit_amount_in_ratio.1)
+            .add_message(mint_msg)
+            .add_attribute("mint_shares_amount", user_shares)
+            .add_attribute("receiver", recipient.as_str()))
+    }
+}
+
+pub fn handle_any_deposit_swap_reply(
+    mut deps: DepsMut,
+    env: Env,
+    data: SubMsgResult,
+) -> Result<Response, ContractError> {
+    // Attempt to directly parse the data to MsgSwapExactAmountInResponse outside of the match
+    let resp: MsgSwapExactAmountInResponse = data.try_into()?;
+
+    let (swap_direction, left_over_amount, recipient, deposit_amount_in_ratio) =
+        CURRENT_SWAP_ANY_DEPOSIT.load(deps.storage)?;
+
+    let pool_config = POOL_CONFIG.load(deps.storage)?;
+
+    // get post swap balances to create positions with
+    let (balance0, balance1): (Uint128, Uint128) = match swap_direction {
+        SwapDirection::ZeroToOne => (
+            left_over_amount,
+            Uint128::new(resp.token_out_amount.parse()?),
+        ),
+        SwapDirection::OneToZero => (
+            Uint128::new(resp.token_out_amount.parse()?),
+            left_over_amount,
+        ),
+    };
+    // Create the position after swapped the leftovers based on swap direction
+    let mut coins_to_mint_for = vec![];
+    if !balance0.is_zero() || !deposit_amount_in_ratio.0.is_zero() {
+        coins_to_mint_for.push(Coin {
+            denom: pool_config.token0.clone(),
+            amount: balance0 + deposit_amount_in_ratio.0,
+        });
+    }
+    if !balance1.is_zero() || !deposit_amount_in_ratio.1.is_zero() {
+        coins_to_mint_for.push(Coin {
+            denom: pool_config.token1.clone(),
+            amount: balance1 + deposit_amount_in_ratio.1,
+        });
+    }
+
+    let (mint_msg, user_shares) = mint_msg_user_shares(
+        deps.branch(),
+        &env,
+        &(coins_to_mint_for[0].amount, coins_to_mint_for[1].amount),
```

Comment:
Would this not fail for single sided positions?, if we don't hit one of the above if cases, we only send an array of length 1. Better would be to work with a tuple here, pushing to an array also gives us the risk of mistaking token0 for token1 and vice versa